more_itertools/more.py

```diff
@@ -137,7 +145,16 @@ def chunked(iterable, n):
     into RAM on the client.
 
     """
-    return iter(partial(take, n, iter(iterable)), [])
+    iterator = iter(partial(take, n, iter(iterable)), [])
+    if strict:
+        def ret(iterator):
+            for chunk in iterator:
+                if len(chunk) != n:
+                    raise ValueError('There are fewer items in the last chunk!')
+                yield chunk
+        return iter(ret(iterator))
```

Comment:
Any particular reason to return this, instead of just getting rid of the function definition and doing `yield chunk`?